static/js/protein/data_processing_utils.ts

```diff
@@ -592,3 +529,271 @@ export function getProteinDescription(data: GraphNodes): string {
+    seen.add(node.name);
+    return !duplicate && node.id !== centerNodeId;
+  });
+
+  // descending order of interaction confidenceScore
+  neighbors.sort((n1, n2) => n2.value - n1.value);
+  // consider only top 10 interactions to avoid clutter
+  neighbors = neighbors.slice(0, MAX_INTERACTIONS);
+
+  const centerDatum = nodeFromId(centerNodeId, 0);
+
+  const linkData: InteractionLink[] = neighbors.map((node) => {
+    return {
+      score: node.value,
+      source: centerNodeId,
+      target: node.id,
+    };
+  });
+
+  return {
+    // guaranteed to be [] because we don't support self-interactions yet
+    linkDataNested: [[], linkData],
+    nodeDataNested: [[centerDatum], neighbors],
+  };
+}
+
+/**
+ * Given response and key, map each datum in response.data to datum[key] and return map.
+ * If response.data is empty, return [].
+ */
+export function getFromResponse<K extends keyof V1BioResponseDatum>(
+  resp: V1BioResponse,
+  key: K
+): V1BioResponseDatum[K][] {
+  if (_.isEmpty(resp.data)) {
+    return [];
+  }
+  return resp.data.map((datum) => datum[key]);
+}
+
+/**
+ * Given a record mapping interaction IDs to corresponding scores, construct and return symmetric score record
+ * such that for each interaction A_B in the input record, both A_B and B_A map to the maximum of the scores of
+ * A_B and B_A if the score of B_A is in the input record, or the score of A_B otherwise. Skip all malformed IDs.
+ *
+ * Since we currently do not graphically distinguish between the A_B and B_A scores,
+ * we choose to have a symmetric score store.
+ *
+ * In the future perhaps we should make the graph directed to make relationships like "A phosphorylates B"
+ * more clear. I also think there should be assays for which the scores are not symmetric -
+ * ex. "knockout A and see what happens to the expression of B". These might not exist in the graph currently
+ * but could very well be imported in the future. Thus, we might have two or more edges between proteins
+ * for each type of interaction they participate in, with different scores for each.
+ */
+export function symmetrizeScoreRec(
+  scoreRec: Record<string, number>
+): Record<string, number> {
+  for (const interactionId of Object.keys(scoreRec)) {
+    const interactionDcid = ppiDcidFromId(interactionId);
+    const proteins = proteinsFromInteractionDcid(ppiDcidFromId(interactionId));
+    if (proteins === null) {
+      console.warn(`Invalid interaction ID ${interactionDcid} -- skipping`);
+      continue;
+    }
+    const [proteinA, proteinB] = proteins;
+    const scoreAB = _.get(
+      scoreRec,
+      `${proteinA}_${proteinB}`,
+      DEFAULT_INTERACTION_SCORE
+    );
+    const scoreBA = _.get(
+      scoreRec,
+      `${proteinB}_${proteinA}`,
+      DEFAULT_INTERACTION_SCORE
+    );
+    const maxScore = Math.max(scoreAB, scoreBA);
+    scoreRec[`${proteinA}_${proteinB}`] = maxScore;
+    scoreRec[`${proteinB}_${proteinA}`] = maxScore;
+  }
+  return scoreRec;
```

Comment:
usually it's better to avoid mutating the object that gets passed in, so here I would create a new record object that holds the new values

Reply:
oh yea I also like pure functions. done!